api/odl.py

```diff
@@ -1040,6 +1038,27 @@ def __init__(self, _db, collection):
 
         self.base_url = collection.external_account_id
 
+    @staticmethod
+    def _parse_feed_from_response(response):
+        """Parse ODL (Atom) feed from the HTTP response.
+
+        :param response: HTTP response
+        :type response: requests.Response
+
+        :return: Parsed ODL (Atom) feed
+        :rtype: dict
+        """
+        response_content = response.content
+
+        if isinstance(response_content, bytes):
+            response_content = response_content.decode("utf-8")
+        elif not isinstance(response_content, str):
+            raise ValueError("Response content must be a string or byte-encoded value")
+
+        feed = feedparser.parse(response_content)
```

Comment:
I thought `feedparser.parse` could handle both `bytes` and `str`. If that's the case, it may be possible to simplify this.

Reply:
Yes, you're right. I think I had some issue back in Python 2.x. Python 3.x handles both cases correctly.